src/flow.c

```diff
@@ -606,8 +606,12 @@ void FlowInitConfig(bool quiet)
             FatalError("Invalid value for flow.hash-size: NULL");
         }
 
-        if (StringParseUint32(&configval, 10, strlen(conf_val), conf_val) > 0 || configval == 0) {
-            flow_config.hash_size = configval;
+        if (StringParseUint32(&configval, 10, strlen(conf_val), conf_val)) {
+            if (configval != 0) {
+                flow_config.hash_size = configval;
+            } else {
+                FatalError("Invalid value for flow.hash-size: 0");
+            }
         } else {
             FatalError("Invalid value for flow.hash-size");
```

Comment:
can't we use this path but add something like "valid range is 1-XXX" ?

Reply:
ok
(not sure MAX_UINT32 which is valid is meaningful...)